### PR TITLE
feat(index): Basic flow support

### PR DESCRIPTION
**Pull request by tamoore**
## Purpose

In discussions over the last few months we've come to the conclusion that we want to add flow support to the style-guide primarily for the developer experience benefits.

## Approach

I looked at this from a load of different angles. Ultimately, I realised that a lib def is not required for the styleguide and the reason that I thought it was came down to a config setting in one of the projects here at SEEK internally. Therefore, whilst it's not a bad idea to also have one, we can just go ahead with defining props as flow types and using a babel transformer to convert those to runtime props.

That's the approach I've taken here.

## Notes.

A couple of things to be aware of. Projects, when updating the styleguide, will need to also update their webpack decorator to [v2.2.0](https://github.com/seek-oss/seek-style-guide-webpack/releases/tag/v2.2.0)

I have a PR for sku which also adds flow type stripping and proptype transformation:
https://github.com/seek-oss/sku/pull/122

### CODE REVIEW

**Comment by markdalgleish** [comment]
Just as an update, had a chat yesterday with @tamoore so I could better understand how this works from a consumer's perspective, and it sounds pretty good. The plan is to add a section to the readme to help guide Flow users in terms of what configuration is necessary on their end, once we've confirmed what the minimal config requirements are.

**Comment by tamoore** [comment]
Have added a README section for this PR. Let me know if anything is unclear.

**Comment by tamoore** [comment]
Fixes #420

File: README.md [hunk 1]
```diff
@@ -330,6 +330,34 @@ Then, include the following in your Babel config:
...
+[lints]
+
+[options]
+# This is required to prevent errors to less file imports
+module.name_mapper.extension='less' -> '<PROJECT_ROOT>/no-declarations.js.flow'
+
+# Good idea to ignore json too
+module.name_mapper.extension='json' -> '<PROJECT_ROOT>/no-declarations.js.flow'
+```
+> *no-declaratins.js.flow* is just an empty file
```
Comment (markdalgleish) [inline README.md]: Should be `no-declarations`.

**Pull request merged**